### ISSUE TITLE
Affix top and bottom problems with sticky header and footer layout

### ISSUE THREAD

**Issue by kspearrin** [twbs/bootstrap#15032]
I am using 3.3.0 with a fluid layout and a fixed column. The fixed column has a menu in it which is affixed. The layout also implements sticky footer and fixed header.

Since version 3.3.0, affix does not behave properly with this layout.

Please see the following fiddle example: http://jsfiddle.net/eqrvupcq/6/embedded/result/  (you may need to adjust browser window size smaller to illustrate scrolling and affixing).

You will notice that when scrolling down to the bottom, the affix sidebar prematurely stops about half way down the page, instead of moving all the way to the defined bottom (footer height).

For comparison, here is the same example with 3.2.0, which behaves as expected: http://jsfiddle.net/eqrvupcq/8/embedded/result/ (you may need to adjust browser window size smaller to illustrate scrolling and affixing).

I am not sure if the layout has anything to do with the problem, but it is illustrated with these examples.

Also note there is a problem with both of these examples (3.3.0 and 3.2.0) when scrolling all the way to the bottom (so that the affix hits the bottom footer) and then moving back up. The affixed element will jump all the way back to the top instead of following the scrolling back up. It is then permanently "broken" from affixing anymore until you refresh the page.

Please let me know if any ore clarification is needed. Thanks.

**Comment by tobscure** [comment]
Same issue here.

**Comment by mdo** [comment]
Punting on this for v4.

**Comment by Xegyn** [comment]
I just ran into this issue myself with a sticky footer, and came across this thread. I think I have the fix here: https://github.com/twbs/bootstrap/pull/15694. 
I found the commit that broke the affix behavior between 3.2.0 and 3.3.0 and I think this accommodates both situations to work. The scrollHeight was changed from the document to the body height, which in the sticky footer's case is the viewport's height. For the sticky footer layout, using the document height yields the correct behavior.
